main.cpp

```diff
@@ -277,36 +166,89 @@ struct HeapA
 
  Wait for my code review.
  */
+
 #include <iostream>
 #include <string>
+#include <cmath>
 
 struct IntType;
 struct FloatType;
 struct DoubleType;
 
 struct FloatType
 {
-    FloatType(float f) : value(new float(f)) { }
+    explicit FloatType(float f) : value(new float(f)) { }
+
     ~FloatType()
     {
         delete value;
         value = nullptr;
     }
-    operator float()
-    {
-        return *value;
-    }
 
     FloatType& add( float rhs );
-   
     FloatType& subtract( float rhs );
-   
     FloatType& multiply( float rhs );
-   
     FloatType& divide( float rhs);
-   
+    operator float() const { return *value; }
+    FloatType& pow(const FloatType&);
+    FloatType& pow(const DoubleType&);
+    FloatType& pow(const IntType&);
+    FloatType& pow(float rhs);
+
     private:
         float* value = nullptr;
+        FloatType& powInternal(const float exp);
```

Comment:
align the `private` keyword with the left edge of the class's curly braces.
The private variables and functions should not be indented further than the public variables and functions
- [ ] revise